python/cudf/cudf/tests/test_concat.py

```diff
@@ -593,3 +593,421 @@ def test_concat_dataframe_with_multiIndex(df1, df2):
+            axis=axis,
+        ),
+    )
+
+
+@pytest.mark.parametrize("ignore_index", [False, True])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [0])
+# leaving out axis = 1 because pandas returns object index not range index
+def test_concat_join_no_overlapping_columns_many_and_empty_axis_0(
+    ignore_index, sort, join, axis
+):
+    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
+    pdf6 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        }
+    )
+    pdf_empty = pd.DataFrame()
+
+    gdf4 = gd.from_pandas(pdf4)
+    gdf5 = gd.from_pandas(pdf5)
+    gdf6 = gd.from_pandas(pdf6)
+    gdf_empty = gd.from_pandas(pdf_empty)
+
+    assert_eq(
+        pd.concat(
+            [pdf4, pdf5, pdf6, pdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+        gd.concat(
+            [gdf4, gdf5, gdf6, gdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+    )
+
+
+# the folloing 3 cases require an index_reset of pandas
+# pandas returns an empty object index instead of a range index.
+# All are cases where there are no overlapping column names
+
+
+@pytest.mark.parametrize("ignore_index", [True])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [0, 1])
+def test_concat_join_no_overlapping_columns_many_and_empty_ignore_index_true(
+    ignore_index, sort, join, axis
+):
+    pdf4 = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    pdf5 = pd.DataFrame({"c": [7, 8, 9], "d": [10, 11, 12]})
+    pdf6 = pd.DataFrame(
+        {
+            "x": range(10),
+            "y": list(map(float, range(10))),
+            "z": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+        }
+    )
+    pdf_empty = pd.DataFrame()
+
+    gdf4 = gd.from_pandas(pdf4)
+    gdf5 = gd.from_pandas(pdf5)
+    gdf6 = gd.from_pandas(pdf6)
+    gdf_empty = gd.from_pandas(pdf_empty)
+
+    assert_eq(
+        pd.concat(
+            [pdf4, pdf5, pdf6, pdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ).reset_index(drop=True),
+        gd.concat(
+            [gdf4, gdf5, gdf6, gdf_empty],
+            sort=sort,
+            join=join,
+            ignore_index=ignore_index,
+            axis=axis,
+        ),
+    )
+
+
+@pytest.mark.parametrize("ignore_index", [False])
+@pytest.mark.parametrize("sort", [True, False])
+@pytest.mark.parametrize("join", ["inner", "outer"])
+@pytest.mark.parametrize("axis", [1])
+def test_concat_join_no_overlapcol_emptydf_ignoreindex_false_axis_1(
```

Comment:
Do we need this test? Why not just add `ignore_index`'s `False` parameter to the test above this one?

Reply:
I put added this case separately because it's one of three cases that require a reset_index of pandas. I added a comment above these three cases to note that pandas returns an object index instead of a range index. Keith mentioned that this might be an issue on Pandas end.

Reply:
If we set `check_index_type` to False on the above tests as you suggested earlier then we don't need to isolate the three cases. I will do that and remove the cases in question.

Reply:
Are we talking about the row indexes or the column indexes?

We generally do our best to match the row index type, except in the situations where Pandas returns an empty object type Index when it could/should be logically using a RangeIndex.

For column index type, there's situations where Pandas returns a RangeIndex where I'm not sure if we support that or if we end up creating a materialized Index of a type instead.

Reply:
In this case here, I was talking about row indexes. Yes, Pandas is returning an empty object index. Setting check_index_type to False will make the tests pass. 

On the other hand in another case, when axis=1, Pandas is returning a RangeIndex for its column index type. I'm not sure what to do about it, which is why I left the case out of the tests.